workitem/link/link_repository_blackbox_test.go

```diff
@@ -855,3 +855,34 @@ func (s *linkRepoBlackBoxTest) TestListChildLinks() {
 		require.True(t, foundAC, "failed to find link A-C")
 	})
 }
+
+func (s *linkRepoBlackBoxTest) TestDeleteLinkAndListChildren() {
+	s.T().Run("delete link and list children", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.WorkItems(2),
+			tf.WorkItemLinksCustom(1, func(fxt *tf.TestFixture, idx int) error {
+				l := fxt.WorkItemLinks[idx]
+				l.LinkTypeID = link.SystemWorkItemLinkTypeParentChildID
+				l.SourceID = fxt.WorkItems[0].ID
+				l.TargetID = fxt.WorkItems[idx+1].ID
+				return nil
+			}),
+		)
+		hasChildren, err := s.workitemLinkRepo.WorkItemHasChildren(s.Ctx, fxt.WorkItems[0].ID)
+		require.NoError(t, err)
+		require.True(t, hasChildren)
+
+		childrenList, totalCount, err := s.workitemLinkRepo.ListWorkItemChildren(s.Ctx, fxt.WorkItems[0].ID, nil, nil)
```

Comment:
These list assertions should be repeated after Delete again (below).

Reply:
done in ae32b71